Use libc and Samba's existing replacements instead of probing host libbsd.
On Linux, rep_getpeereid uses SO_PEERCRED. Process-title replacement is a no-op.

--- a/lib/replace/wscript
+++ b/lib/replace/wscript
@@ -443,12 +443,7 @@
 
     strlcpy_in_bsd = False
 
-    # libbsd on some platforms provides strlcpy and strlcat
-    if not conf.CHECK_FUNCS('strlcpy strlcat'):
-        if conf.CHECK_FUNCS_IN('strlcpy strlcat', 'bsd', headers='bsd/string.h',
-                               checklibc=True):
-            strlcpy_in_bsd = True
-    elif conf.env.enable_fuzzing:
+    if conf.CHECK_FUNCS('strlcpy strlcat') and conf.env.enable_fuzzing:
         # Just to complicate it more, some versions of Honggfuzz have
         # got strlcpy and strlcat in libc, but not in <string.h>
         # (unless it is there coincidentally, on a BSD). Therefore we
@@ -459,15 +454,11 @@
         # skip the check when not in fuzzing mode.
         conf.CHECK_HEADERS('bsd/string.h')
 
-    if not conf.CHECK_FUNCS('getpeereid'):
-        conf.CHECK_FUNCS_IN('getpeereid', 'bsd', headers='sys/types.h bsd/unistd.h')
-    if not conf.CHECK_FUNCS_IN('setproctitle', 'setproctitle', headers='setproctitle.h'):
-        conf.CHECK_FUNCS_IN('setproctitle', 'bsd', headers='sys/types.h bsd/unistd.h')
-    if not conf.CHECK_FUNCS('setproctitle_init'):
-        conf.CHECK_FUNCS_IN('setproctitle_init', 'bsd', headers='sys/types.h bsd/unistd.h')
+    conf.CHECK_FUNCS('getpeereid')
+    conf.CHECK_FUNCS_IN('setproctitle', 'setproctitle', headers='setproctitle.h')
+    conf.CHECK_FUNCS('setproctitle_init')
 
-    if not conf.CHECK_FUNCS('closefrom'):
-        conf.CHECK_FUNCS_IN('closefrom', 'bsd', headers='bsd/unistd.h')
+    conf.CHECK_FUNCS('closefrom')
 
     conf.CHECK_CODE('''
                 struct ucred cred;
